documentation/CAMARA-API-Design-Guide.md

```diff
@@ -0,0 +1,1165 @@
+
+* `resource` is optional. For APIs with several `paths`, it may include the resource in the path. For example, from `/qod/v1/sessions/{sessionId}`, it would be `sessions`.
+
+* `action`: There are two cases:
+  - For POST operations with a verb, it will be the verb. For example, from `POST /location-verification/v1/verify`, it would be `verify`.
+    - For endpoints designed as POST but with underlying logic retrieving information, a CRUD action `read` may be added. However, if it is a path with a single operation, and it is not expected to have more operations on it, the CRUD action is not necessary.
+  - For CRUD operations on a resource in paths, it will be one of:
+    - `create`: For operations creating the resource, typically `POST`.
+    - `read`: For operations accessing to details of the resource, typically `GET`.
+    - `update`: For operations modifying the resource, typically `PUT` or `PATCH`.
+    - `delete`: For operations removing the resource, typically `DELETE`.
+    - `write` : For operations creating or modifying the resource, when differentiation between `create` and `update` is not needed.
+
+* Eventually we may need to add a level to the scope, such as `api-name:[resource:]action[:detail]`, to deal with cases when only a set of parameters/information has to be allowed to be returned. Guidelines should be enhanced when those cases happen.
+
+
+##### Examples
+
+| API                   | path          | method | scope                                             |
+|-----------------------|---------------|--------|---------------------------------------------------|
+| location-verification | /verify       | POST   | `location-verification:verify`                    |
+| qod                   | /sessions     | POST   | `qod:sessions:create`, or<br>`qod:sessions:write` |
+| qod                   | /qos-profiles | GET    | `qod:qos-profiles:read`                           |
+
+
+#### 6.6.2. API-level scopes (sometimes referred to as wildcard scopes in CAMARA)
+
+The decision on the API-level scopes was made within the [Identity and Consent Management Working Group](https://github.com/camaraproject/IdentityAndConsentManagement) and is documented in the design guidelines to ensure the completeness of this document. 
+The scopes will always be those defined in the API Specs YAML files. Thus, a scope would only provide access to all endpoints and resources of an API if it is explicitly defined in the API Spec YAML file and agreed in the corresponding API subproject. 
+
+### 6.7. Resource access restriction
+
+In some CAMARA APIs there are functions to create resource (via POST) and then later query them via id and/or list (with GET) or delete them (via DELETE). For example we have sessions, payments, subscriptions, etc..
+
+For the GET and DELETE operations we must restrict the resource(s) targeted depending on the information provided in the request. Basically we consider 2 filters:
+* API client (aka ClientId)

```

Comment:
Consistency: Would be good to use either _API client_  or _API consumer_, but not both. 

Personally, _API consumer_  or _API invoker_ clearer.